### PR TITLE
CONTRIBUTING: update release checklist

### PR DESCRIPTION
**Pull request by intelliot**
## High Level Overview of Change

Merges in all of the still-potentially-relevant content from:

https://github.com/XRPLF/xrpl.js/wiki/Release-Checklist

### Context of Change

The original release checklist was [here](https://github.com/XRPLF/xrpl.js/wiki/Release-Checklist), but it probably wasn't consulted or updated when the release process docs were moved to `CONTRIBUTING.md`.

`--tag beta` was already documented there but was missing from `CONTRIBUTING.md` for some time (it's already there now).

To ensure nothing falls through the cracks, this PR adopts content into `CONTRIBUTING.md`. After this is merged, the wiki page will be deprecated and replaced with a link to `CONTRIBUTING.md`.

### Type of Change

- [x] Documentation Updates

## Future Tasks

After this is merged, the wiki page will be deprecated and replaced with a link to `CONTRIBUTING.md`.

### CODE REVIEW

**Review by khancode — APPROVED** [review]

File: CONTRIBUTING.md
Comment (JST5000) [inline CONTRIBUTING.md:203]: Re-adding numbers using auto numbering since checkboxes aren't dynamic, and with all the alt-tabbing needed it's easy to lose track of where you are unless you have the numbers. 
```suggestion
1. Checkout `main` (or your beta branch) and `git pull`.
1. Create a new branch (`git checkout -b <BRANCH_NAME>`) to capture updates that take place during this process.
1. Update `HISTORY.md` to reflect release changes.

- [ ] Update the version number and release date, and ensure it lists the changes since the previous release.

1. Run `npm run docgen` if the docs were modified in this release to update them (skip this step for a beta).
1. Run `npm run build` to triple check the build still works
1. Run `npx lerna version --no-git-tag-version` - This creates a draft PR and bumps the versions of the packages.

- [ ] For each changed package, pick what the new version should be. Lerna will bump the versions, commit version bumps to `main`, and create a new git tag for each published package.
- [ ] If publishing a beta, make sure that the versions are all of the form `a.b.c-beta.d`, where `a`, `b`, and `c` are identical to the last normal release except for one, which has been incremented by 1.

1. Run `npm i` to update the package-lock with the updated versions.
1. Create a new PR from this branch into `main` and merge it (you can directly merge into the beta branch for a beta).
1. Checkout `main` and `git pull` (you can skip this step for a beta since you already have the latest version of the beta branch).
1. Actually publish the packages with one of the following:

- [ ] Stable release: Run `npx lerna publish from-package --yes`
- [ ] Beta release: Run `npx lerna publish from-package --dist-tag beta --yes`
      Notice this allows developers to install the package with `npm add xrpl@beta`
- [ ] If requested, enter your [npmjs.com](https://npmjs.com) OTP (one-time password) to complete publication.

1. If not a beta release: Create a new branch (`git checkout -b <BRANCH_NAME>`) to capture the updated packages from the release. Merge those changes into `main`.
```

**Comment by JST5000** [comment]
Opened this PR to update this branch with numbers instead of checkboxes - will merge if that seems worthwhile :) - https://github.com/XRPLF/xrpl.js/pull/2239

**Comment by JST5000** [comment]
@intelliot thoughts ^?

**Comment by intelliot** [comment]
Doesn't matter to me. Checkboxes make sense conceptually, i.e. you check off one item before moving to the next. But if you prefer numbering, that's fine too.

**Comment by intelliot** [comment]
Just an idea: If you're going to alt-tab and want to keep track of where you are, you could copy-paste the checklist into a GitHub issue or PR description, and check off the boxes as you go along.

File: CONTRIBUTING.md
Comment (mvadari) [inline CONTRIBUTING.md:174]: IMO this should be labeled as the process for a PR instead of for a release

File: CONTRIBUTING.md
Comment (intelliot) [reply CONTRIBUTING.md:237]: for the npm package? It looks like maintainers are:

ahbritto
justmoon
geertweening
wilsonianb
shekenahglory
intelliot
hansbergren

Unfortunately, it doesn't look like I have permission to add maintainers. Though I could be wrong about this, as the npmjs interface isn't clear.

File: CONTRIBUTING.md
Comment (intelliot) [inline CONTRIBUTING.md:174]:
```suggestion
## PR process
```

File: CONTRIBUTING.md
Comment (mvadari) [reply CONTRIBUTING.md:237]: Yes, I remember that being an issue the last time we dealt with this.

Comment (intelliot) [reply CONTRIBUTING.md:237]: Alright, I'll add my name in the doc for now

Comment (intelliot) [inline CONTRIBUTING.md:237]:
```suggestion
For ripple-lib:

 - Have one of the ripple-lib package maintainers push to `ripple-lib` (npm package name). You can contact [@intelliot](https://github.com/intelliot) to request the npm publish.
- For ripple-lib releases, cross-publish the package to `xrpl` with `--tag ripple-lib`
  - [Here's why](https://blog.greenkeeper.io/one-simple-trick-for-javascript-package-maintainers-to-avoid-breaking-their-user-s-software-and-to-6edf06dc5617).
```

**Comment by intelliot** [comment]
I think this is ready for review @JST5000 @mvadari

**Review by JST5000 — APPROVED** [review]

**Pull request merged**
